specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/devices.json

```diff
@@ -0,0 +1,582 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Security Center",
+    "description": "API spec for Microsoft.Security (Azure Security Center) resource provider",
+    "version": "2019-08-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/devices": {
+      "get": {
+        "x-ms-examples": {
+          "Get Devices": {
+            "$ref": "./examples/Devices/GetDevicesList.json"
+          }
+        },
+        "tags": [
+          "Device Inventory"
+        ],
+        "description": "Get list of the devices.",
+        "operationId": "Devices_List",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/SolutionName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DeviceList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/devices/{deviceId}": {
```

Comment:
how is a device created? Is there a way to remove a device from under a solution?

Reply:
Device is not created via REST API, it is created in our internal pipelines. We are also not allowing to delete it via the REST API

Reply:
Are these proxy resources? Why do you need GET through ARM? What is the PATCH scenario?

Reply:
Yes, this is a proxy resource. We are implementing device inventory functionality. There are sensors scanning the network and discovering devices that are saved in our data-base. The user uses GET in order to see the full device inventory list of all his devices. There are certain fields that the user is allowed to edit (like device name, type, operating system) and then he uses the PATCH. Most of the fields the user is not allowed to edit.